src/v/cluster/partition.cc

```diff
@@ -321,8 +321,9 @@ ss::future<result<kafka_result>> partition::replicate(
     if (!res) {
         co_return ret_t(res.error());
     }
-    co_return ret_t(kafka_result{
-      kafka::offset(_translator->from_log_offset(res.value().last_offset)())});
+    co_return ret_t(
```

Comment:
Do I understand correctly that this commit isn't necessary because the issue is fixed in previous commit but it's nice to have as a safety measure?

Reply:
exactly